### PR TITLE
Add WCS GetCapabilities and DescribeCoverage parsers

### LINKED ISSUE #485: WCS capabilities parsing
Closes #482

Establish support for parsing a WCS capabilities document, versions 1.0.0 and 2.0.1, and support for representing that object model in JavaScript for use in WebWorldWind.

### PR DESCRIPTION
**Pull request by zglueck**
### Description of the Change

This change adds an XML parser which produces simple javascript object models for the Web Coverage Service (WCS) GetCapabilities and DescribeCoverage responses. It supports version 1.0.0 and 2.0.1 of the WCS specification.

### Applicable Issues

Closes #485

### CODE REVIEW

**Comment by markpet49** [comment]
@zglueck Hi Zach. There are a lot of items missing docs. Are those coming later or are these classes internal use only and so docs aren't so critical?

File: src/ogc/gml/GmlRectifiedGrid.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            origin.show = element.getAttribute("xlink:show");
+            origin.actuate = element.getAttribute("xlink:actuate");
+            origin.nilReason = element.getAttribute("nilReason");
+            origin.remoteSchema = element.getAttribute("gml:remoteSchema");
+            origin.owns = element.getAttribute("owns");
+
+            var children = element.children || element.childNodes;
+
+            for (var c = 0; c < children.length; c++) {
+                var child = children[c];
```
Comment (markpet49) [inline src/ogc/gml/GmlRectifiedGrid.js]: Dave pointed out the other day that the form:
for (var c = 0, len=children.length; c < len; c++)
is preferable over:
for (var c = 0; c < children.length; c++)
for performance reasons.

**Comment by zglueck** [comment]
@markpet49 all of the methods were designed with the intention of being internal use only at this point.

File: src/ogc/gml/GmlRectifiedGrid.js (hunk 1 above)
Comment (zglueck) [reply src/ogc/gml/GmlRectifiedGrid.js]: I agree, and usually use that approach with for loops; however, the legacy (WMS, WMTS) XML parsing code uses the pattern applied in WCS. I opted for consistency over performance.

**Comment by markpet49** [comment]
@zglueck @pdavidc Is the criteria for referencing classes in WorldWind.js that they be public?

**Comment by markpet49** [comment]
@zglueck Hi Zach. This looks like well written parsing code to me.... Two questions:
1. Is attribute case sensitivity an issue? e.g. child.localName === "lonLatEnvelope" vs. child.localName === "lonlatenvelope" Perhaps the relevant specs require certain casing?
2. Is there a need or use for a check to make sure the document being parsed is well formed. i.e. not missing certain required attributes?

**Comment by zglueck** [comment]
@markpet49 

1. Yes, element and attribute names are case sensitive. Is there a goof somewhere?

2. We've discussed checking certain document elements in the past and settled on a "not my job to verify your server output" approach :) Technically, applying higher level document check wouldn't be too tough.

**Comment by markpet49** [comment]
1. No goofs that I saw, I've just dealt with parsing a lot of stuff over the years and always had to deal with data producers who would case attribute names however they felt like. Strangely, many of them were government entities. ;-)
2. Sounds fine to me, I'm a still a bit fuzzy on the use cases for these classes.

**Comment by markpet49** [comment]
I have no further feedback, I'll leave this open for @pdavidc to look over.

**Comment by zglueck** [comment]
@pdavidc thanks for the notes.

1. IE11 compatibility is supported throughout
2. Updated the naming
3. Simplified the logging by removing missingXmlDom and changing unsupportedWcsVersion to unsupportedVersion
4. Added the requisite internal use only documentation and enhanced the public facing documentation

* added WcsCapabilities and WcsDescribeCoverage to the WorldWind object

**Pull request merged**